### PR TITLE
Remove use of ext 

### PR DESCRIPTION
**Pull request by jd-lara**
related to #9 

This PR doesn't introduce logic changes

cc. @daniel-thom

### CODE REVIEW

File: src/base/simulation.jl [hunk 1]
```diff
@@ -398,38 +395,24 @@ function _index_dynamic_system!(sys::PSY.System)
...
+    dyn_lines = !isempty(PSY.get_components(PSY.DynamicBranch, sys)),
+    global_index = global_state_index,
+    voltage_buses_ix = voltage_buses_ix,
+    current_buses_ix = current_buses_ix,
+    total_shunts = total_shunts,
+    global_vars = global_vars,
+    lookup = lookup,
+    DAE_vector = DAE_vector)
+
+    @assert get_ω_sys(inputs) != -1
```
Comment (rodrigomha) [inline src/base/simulation.jl:411]: Should we provide an error comment here? Such as no device was found in REF bus or similar?

File: src/base/simulation.jl [hunk 2]
```diff
@@ -148,17 +148,14 @@ function Simulation(
...
-    aux_arrays = Dict(
-        1 => collect(zeros(T, bus_count)),                       #I_injections_r
-        2 => collect(zeros(T, bus_count)),                       #I_injections_i
-        3 => collect(zeros(T, get_n_injection_states(system))),  #injection_ode
-        4 => collect(zeros(T, get_n_branches_states(system))),   #branches_ode
-        5 => collect(zeros(Complex{T}, bus_count)),              #I_bus
-        6 => collect(zeros(T, 2 * bus_count)),                   #I_balance
-    )
-    system.internal.ext[AUX_ARRAYS] = aux_arrays
+    get_aux_arrays(inputs)[1] = collect(zeros(T, bus_count)),                       #I_injections_r
```
Comment (rodrigomha) [inline src/base/simulation.jl:153]: Remove the commas here

File: src/base/simulation.jl (hunk 1 above)
Comment (jd-lara) [reply src/base/simulation.jl:411]: The error should have been caught earlier. This is a check afterward.

File: Project.toml [hunk 3]
```diff
@@ -20,5 +20,5 @@ DocStringExtensions = "~0.8.2"
 ForwardDiff = "~v0.10"
 InfrastructureSystems = "~0.15"
 NLsolve = "4"
-PowerSystems = "~0.25"
+PowerSystems = "~0.26"
```
Comment (rodrigomha) [inline Project.toml:23]: Do we need PSY 0.26? Is it possible to use 0.25 or at least give compatibility to 0.25? After this release I would like to add the SIIPExamples that are using 0.25.

File: src/base/definitions.jl [hunk 4]
```diff
@@ -90,20 +90,10 @@ const Q_ref_index = 4
...
 const CONTROL_REFS = "control_refs"
-const GLOBAL_VARS = "global_vars"
-const VOLTAGE_BUSES_IX = "voltage_buses_ix"
-const CURRENT_BUSES_IX = "current_buses_ix"
-const TOTAL_SHUNTS = "total_shunts"
-const AUX_ARRAYS = "aux_arrays"
-const LOOKUP = "lookup"
-const DYN_LINES = "dyn_lines"
 
 const SIMULATION_ACCEPTED_KWARGS = [:initial_guess, :initialize_simulation, :system_to_file]
```
Comment (rodrigomha) [inline src/base/definitions.jl:99]: We should eliminate initial_guess here.

File: src/base/simulation_initialization.jl [hunk 5]
```diff
@@ -1,25 +1,29 @@
-function calculate_initial_conditions!(sys::PSY.System, initial_guess::Vector{Float64})
+function calculate_initial_conditions!(
+    inputs::SimulationInputs,
+    initial_guess::Vector{Float64},
```
Comment (rodrigomha) [inline src/base/simulation_initialization.jl:3]: Are we using here initial_guess in some way?

File: Project.toml (hunk 3 above)
Comment (jd-lara) [reply Project.toml:23]: PSI is already in 0.26 SIIPExamples will have to move all together

File: src/base/definitions.jl (hunk 4 above)
Comment (jd-lara) [reply src/base/definitions.jl:99]: That should be done in a separate PR

File: src/base/simulation_initialization.jl (hunk 5 above)
Comment (jd-lara) [reply src/base/simulation_initialization.jl:3]: Let's do that in a separate PR

File: src/base/simulation.jl [hunk 6]
```diff
@@ -9,6 +8,7 @@ mutable struct Simulation
     callbacks::DiffEqBase.CallbackSet
     solution::Union{Nothing, DiffEqBase.DAESolution}
     simulation_folder::String
+    simulation_inputs::SimulationInputs
     ext::Dict{String, Any}
```
Comment (daniel-thom) [inline src/base/simulation.jl:12]: Can `ext` be removed from the struct?

File: src/base/simulation.jl [hunk 7]
```diff
@@ -284,7 +281,7 @@ function _add_states_to_global!(
     return
 end
 
-function _index_dynamic_system!(sys::PSY.System)
+function make_simulation_inputs(sys::PSY.System)
```
Comment (daniel-thom) [inline src/base/simulation.jl:284]: This feels like a constructor for `SimulationInputs`.

**Pull request merged**